presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -470,6 +481,37 @@ public CounterStat getHighMemorySplitSource()
         return concat(partitionBatches);
     }
 
+    @VisibleForTesting
+    static Optional<Set<HiveColumnHandle>> mergeRequestedAndPredicateColumns(Optional<Set<HiveColumnHandle>> requestedColumns, Set<HiveColumnHandle> predicateColumns)
+    {
+        if (!requestedColumns.isPresent() || predicateColumns.isEmpty()) {
```

Comment:
@mayankgarg1990 Is this the right logic? For a `SELECT count(*) FROM t WHERE a > 10` we can have requestedColumns empty and predicateColumns containing "a". In this case, don't we need to make sure user has access to "a" ?

Reply:
In this case - I am checking for `requestedColumns.isPresent()` and the situation you are talking about will actually be `requestedColumns.get().isEmpty()`.

So this check mainly ensures that due to some reason if we don't end up getting the requestedColumns field populated at all, the encryption provider will now try to get the credentials for all columns.